db.go

```diff
@@ -104,8 +128,32 @@ func (db *DB) SignUp(authData *Auth) (string, error) {
 	return *token.Result, nil
 }
 
-// SignIn is a helper method for signing in a user.
-func (db *DB) SignIn(authData *Auth) (string, error) {
+// SignIn signs in an existing user.
+//
+// The authData parameter can be either:
+//   - An Auth struct
+//   - A map[string]any with keys like: "namespace", "database", "scope", "user", "pass"
+//
+// Example with struct:
+//
+//	db.SignIn(Auth{
+//	  Namespace: "app",
+//	  Database: "app",
+//	  Scope: "user",
+//	  Username: "yusuke",
+//	  Password: "VerySecurePassword123!",
+//	})
+//
+// Example with map:
+//
+//	db.SignIn(map[string]any{
+//	  "namespace": "app",
+//	  "database": "app",
+//	  "scope": "user",
+//	  "user": "yusuke",
+//	  "pass": "VerySecurePassword123!",
```

Comment:
```suggestion
//	  "NS": "app",
//	  "DB": "app",
//	  "user": "yusuke",
//	  "pass": "VerySecurePassword123!",
```

as per https://surrealdb.com/docs/surrealdb/integration/rpc#parameters-20